engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -611,13 +618,8 @@ private BeanGroupProcessedUnit(Object bean, Class<?> group) {
 
 		@Override
 		public boolean equals(Object o) {
-			if ( this == o ) {
-				return true;
-			}
```

Comment:
Do we really gain something by removing this? If not I would vote for keeping it (and update the comment below).